### PR TITLE
Remove or in test suite

### PR DESCRIPTION
**Pull request by canyon289**
Makes things less confusing

### CODE REVIEW

File: pymc/tests/test_distributions_random.py [hunk 1]
```diff
@@ -428,9 +428,9 @@ class TestFlat(BaseTestDistributionRandom):
     ]
 
     def check_rv_inferred_size(self):
-        sizes_to_check = self.sizes_to_check or [None, (), 1, (1,), 5, (4, 5), (2, 4, 2)]
```
Comment (twiecki) [inline pymc/tests/test_distributions_random.py:431]: But this will override `self.sizes_to_check` which the previous code didn't do, is that safe?

Comment (canyon289) [reply pymc/tests/test_distributions_random.py:431]: That's of point of the or statement isnt it? So if it gets overridden already through a conditional why not just override it explicitly in this function?

The or statement makes the test much more confusing to read, the tests pass without it and so far no one has mentioned a reason why its needed

Comment (canyon289) [reply pymc/tests/test_distributions_random.py:431]: Ok I see what youre saying. If its the assignment on self that`s concerning I'll change the PR to purely work off of local variables. All I care about is removing the `or` which doesn't add anything, just makes it harder to trace this test and understand what its doing.

In that case we can change it to the following
```
        sizes_to_check = [None, (), 1, (1,), 5, (4, 5), (2, 4, 2)]
        sizes_expected = [(), (), (1,), (1,), (5,), (4, 5), (2, 4, 2)]
        for size, expected in zip(sizes_to_check, sizes_expected):
            pymc_rv = self.pymc_dist.dist(**self.pymc_dist_params, size=size)
            expected_symbolic = tuple(pymc_rv.shape.eval())
            assert expected_symbolic == expected
```

Comment (ricardoV94) [reply pymc/tests/test_distributions_random.py:431]: This is changing the original behavior, and ignoring when a test uses custom sizes_to_check

Comment (ricardoV94) [reply pymc/tests/test_distributions_random.py:431]: yeah that looks good

**Comment by canyon289** [comment]
done

**Review by ricardoV94 — APPROVED** [review]

**Review by twiecki — APPROVED** [review]

**Pull request merged**